<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <arg name="simulation" default="$(optenv SIMULATION false)" />
  <arg unless="$(arg simulation)" name="config_file" value="$(find mrs_bumper)/config/bumper_params_realsense.yaml"/>
  <arg     if="$(arg simulation)" name="config_file" value="$(find mrs_bumper)/config/bumper_params_sim.yaml"/>
  <arg unless="$(arg simulation)" name="depthmap_topic" value="/$(arg uav_name)/rs_d435/aligned_depth_to_color/image_raw"/>
  <arg     if="$(arg simulation)" name="depthmap_topic" value="/$(arg uav_name)/realsense/camera/depth/image_raw"/>
  <arg unless="$(arg simulation)" name="depth_cinfo_topic" value="/$(arg uav_name)/rs_d435/aligned_depth_to_color/camera_info"/>
  <arg     if="$(arg simulation)" name="depth_cinfo_topic" value="/$(arg uav_name)/realsense/camera/depth/camera_info"/>

  <arg name="ignore_mask" default="$(optenv IGNORE_MASK true)" />
  <arg unless="$(arg ignore_mask)" name="mask_filename" value="$(find mrs_bumper)/masks/realsense_f550_mask.bmp"/>
  <arg     if="$(arg ignore_mask)" name="mask_filename" value=""/>

  <arg name="standalone" default="true" />
  <arg name="manager" default="detection_nodelet_manager" />
  <arg name="nodelet" value="mrs_bumper/Bumper" />
  <arg unless="$(arg standalone)" name="nodelet_load" value="load $(arg nodelet) $(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_load" value="standalone $(arg nodelet)"/>

  <group ns="$(arg uav_name)">
    
    <node pkg="nodelet" type="nodelet" name="mrs_bumper" args="$(arg nodelet_load)" output="screen" launch-prefix="$(arg launch_prefix)">
      
      <param name="uav_name" value="$(arg uav_name)" />
      <param name="path_to_mask" value="$(arg mask_filename)" />
      <param name="frame_id" value="fcu_$(arg uav_name)_horizontal" />
      <rosparam file="$(arg config_file)" />
    
      <remap from="~depthmap" to="$(arg depthmap_topic)" />
      <remap from="~depth_camera_info" to="$(arg depth_cinfo_topic)" />

    </node>
  
  </group>

</launch>

